util/git/client.go

```diff
@@ -335,7 +335,7 @@ func (m *nativeGitClient) Fetch(revision string) error {
 
 	var err error
 	if revision != "" {
-		err = m.runCredentialedCmd("git", "fetch", "origin", revision, "--tags", "--force")
+		err = m.runCredentialedCmd("git", "fetch", "origin", revision)
```

Comment:
Can you please clarify why this change is necessary? Can we rollback this line please unless it is really needed?

Reply:
No need to fetch tags on explicit revision. This reduce a bit the pack size on large repository with many tags.

Feel free to revert the change though.

Reply:
Sounds good. Reverted this change. The rest looks good! Merging once CI is green.